script.js

```diff
@@ -0,0 +1,942 @@
+        innerElement.textContent = text;
+      } else {
+        key.classList.add(classes.KEY_UPPERCASE);
+        innerElement.textContent = text.toUpperCase();
+      }
+    }
+  });
+};
+
+/**
+ * Changes the language stored in localStorage to the opposite.
+ * Changes button's with className 'btn' innerText with name of the language.
+ * Changes keys' innerText to the opposite
+ */
+function changeLanguage() {
+  // 1. remove key-uppercase before update
+  elements.keys.forEach((key) => {
+    key.classList.remove(classes.KEY_UPPERCASE);
+  });
+
+  // 2. changeLanguage
+  // make changes with localStorage
+  // and get source alphabet pair map {'en':'ru'} or {'ru':'en'}
+  let alphabet;
+  let numpad;
+  switch (localStorage.getItem(variables.KEYBOARD_LANGUAGE)) {
+    case variables.languages.RU:
+      localStorage.setItem(variables.KEYBOARD_LANGUAGE, variables.languages.EN);
+      [alphabet, numpad] = getAlphabet(variables.languages.RU);
+      break;
+    case variables.languages.EN:
+    default:
+      localStorage.setItem(variables.KEYBOARD_LANGUAGE, variables.languages.RU);
+      [alphabet, numpad] = getAlphabet(variables.languages.EN);
+      break;
+  }
+
+  // change symbols from english to russian, and backwards
+  changeKeysInnerText(alphabet, numpad);
+
+  // 3. restore uppercase
+  if (isUppercase()) changeCase();
+}
+
+const playKeypressSound = () => {
+  const audio = new Audio('assets/sound/key-press.mp3');
+  audio.volume = 0.5;
+  audio.autoplay = true;
+};
+
+initLanguageFromStorage(); // set language from storage init
+
+function handlerKeyInput(elem, event) {
+  if (!elem) return;
+  const text = elements.textarea;
+  text.focus();
+
+  const isMouse = () => event.type === 'mousedown';
+
+  const el = elem.firstElementChild;
+  switch (el.innerText) {
+    case variables.specialKeys.ESC: case variables.specialKeys.F1: case variables.specialKeys.F2:
+    case variables.specialKeys.F3: case variables.specialKeys.F4: case variables.specialKeys.F6:
+    case variables.specialKeys.F7: case variables.specialKeys.F8: case variables.specialKeys.F9:
+    case variables.specialKeys.F10: case variables.specialKeys.F12:
+    case variables.specialKeys.PRINT_SCREEN: case variables.specialKeys.SCROLL_LOCK:
+    case variables.specialKeys.PAUSE: case variables.specialKeys.INSERT:
+    case variables.specialKeys.NUM_LOCK: case variables.specialKeys.CTRL:
+    case variables.specialKeys.FN: case variables.specialKeys.META:
+      break;
+    case variables.specialKeys.ALT:
+      if (!isMouse()) event.preventDefault();
+      break;
+    case variables.specialKeys.ALTGR:
+      if (!isMouse()) event.preventDefault();
+      break;
+    case variables.specialKeys.F5: {
+      if (isMouse()) document.location.reload();
+      break;
+    }
+    case variables.specialKeys.F11: {
+      if (isMouse()) {
+        if (document.fullscreenElement) {
+          document.exitFullscreen();
```

Comment:
Nested if is bad. Try to optimize it.

Reply:
I rewrote it like this:
```js
      if (isMouse() && document.fullscreenElement) {
        document.exitFullscreen();
      } else if (isMouse() && !document.fullscreenElement) {
        document.documentElement.requestFullscreen();
      }
```
but I think it's also bad :)